packages/common/src/api/tan-query/useSearchResults.ts

```diff
@@ -58,51 +71,56 @@ const getMinMaxFromBpm = (bpm?: string) => {
 }
 
 export const getSearchResultsQueryKey = ({
-  currentUserId,
   query,
   category,
-  limit,
-  offset,
-  source,
   sortMethod,
-  disableAnalytics,
+  pageSize,
   ...filters
-}: SearchArgs) => [
+}: Omit<SearchArgs & { category: SearchCategory }, 'currentUserId'>) => [
   QUERY_KEYS.search,
+  category,
   query,
-  { category, limit, offset, source, sortMethod, disableAnalytics },
-  { ...filters }
+  { sortMethod, pageSize, ...filters }
 ]
 
-export const useSearchResults = (
+export const SEARCH_PAGE_SIZE = 12
+
+const useSearchQueryProps = (
   {
-    currentUserId,
-    query,
+    query = '',
     category,
-    limit,
-    offset,
     source = 'search results page',
     sortMethod,
     disableAnalytics,
+    pageSize = SEARCH_PAGE_SIZE,
     ...filters
-  }: SearchArgs,
+  }: SearchArgs & { category: SearchCategory },
   options?: QueryOptions
 ) => {
+  const { data: currentUserId } = useCurrentUserId()
+  const queryKeyArgs = {
+    query,
+    category,
+    sortMethod,
+    pageSize,
+    ...filters
+  }
   const { audiusSdk, getFeatureEnabled, analytics } = useAudiusQueryContext()
   const queryClient = useQueryClient()
   const dispatch = useDispatch()
 
-  return useQuery({
-    queryKey: getSearchResultsQueryKey({
-      currentUserId,
-      query,
-      category,
-      limit,
-      offset,
-      sortMethod,
-      ...filters
-    }),
-    queryFn: async () => {
+  return {
+    initialPageParam: 0,
+    queryKey: getSearchResultsQueryKey(queryKeyArgs),
+    queryFn: async ({
+      pageParam
+      // unfortunately tanquery was having type issues with pageParam here
```

Comment:
sad :(

happy to follow up on this together if you want to pair and experiment. ok with merging as is though